modules/artifact/src/main/java/org/mule/module/artifact/classloader/JMSResourceReleaser.java

```diff
@@ -0,0 +1,182 @@
+    }
+  }
+
+  /**
+   * Method that returns if the actual thread is the thread that we want to delete this name have to be -ActiveMQ
+   * InactivityMonitor ReadCheckTimer- And make other assertions to know is the thread belong to the current classloader.
+   */
+  private boolean isActiveMQInactivityMonitorTimerThread(ClassLoader classLoader, Thread thread) {
+
+    if (isActiveMQInactivityMonitorTimerThreadTestContext(classLoader, thread)) {
+      return true;
+    }
+    if (!(classLoader instanceof ArtifactClassLoader)) {
+      return false;
+    }
+    String artifactId = ((ArtifactClassLoader) classLoader).getArtifactId();
+    boolean isActiveMqThread = thread.getName().equals(ACTIVEMQ_DRIVER_TIMER_THREAD_NAME);
+
+    return thread.getClass().getSimpleName().equals(ACTIVEMQ_DRIVER_TIMER_THREAD_CLASS_NAME)
+        && isActiveMqThread
+        && (isThreadLoadedByDisposedApplication(artifactId, thread.getContextClassLoader())
+            || isThreadLoadedByDisposedDomain(artifactId, thread.getContextClassLoader()));
+  }
+
+  /**
+   * Test Context: threads are loaded from a classloader type - AppClassLoader - Normal Context: loaded by - CompositeClassLoader
+   */
+  private boolean isActiveMQInactivityMonitorTimerThreadTestContext(ClassLoader classLoader, Thread thread) {
+
+    String artifactId = ((ArtifactClassLoader) classLoader).getArtifactId();
+    return artifactId.equals(TEST_CLASSLOADER_ARTIFACT_ID)
+        && thread.getContextClassLoader().getClass().getSimpleName().equals(CLASSLOADER_CLASS_TEST_CONTEXT)
+        && thread.getName().equals(ACTIVEMQ_DRIVER_TIMER_THREAD_NAME);
+  }
+
+  /**
+   * To ensure that tha thread belong to the current domain
+   */
+  private boolean isThreadLoadedByDisposedDomain(String undeployedArtifactId, ClassLoader threadContextClassLoader) {
+    try {
+      Class threadContextClassLoaderClass = threadContextClassLoader.getClass();
+      if (!threadContextClassLoaderClass.getSimpleName().equals(COMPOSITE_CLASS_LOADER_CLASS_NAME)) {
+        return false;
+      }
+
+      Method getDelegateClassLoadersMethod = threadContextClassLoaderClass.getMethod("getDelegates");
+      List<ClassLoader> classLoaderList = (List<ClassLoader>) getDelegateClassLoadersMethod.invoke(threadContextClassLoader);
+
+      for (ClassLoader classLoaderDelegate : classLoaderList) {
+        if (classLoaderDelegate instanceof ArtifactClassLoader) {
+          ArtifactClassLoader artifactClassLoader = (ArtifactClassLoader) classLoaderDelegate;
+          if (artifactClassLoader.getArtifactId().contains(undeployedArtifactId)) {
+            return true;
+          }
+        }
+      }
+
+    } catch (Exception e) {
+      logger.warn("Exception occurred while attempting to compare {} and {} artifactId.", threadContextClassLoader,
+                  this.getClass().getClassLoader());
+    }
+    return false;
+  }
```

Comment:
This code duplicates the code in the JDBCResourceReleaser. Please refactor for reuse.